File: 0056-merge-intervals/solution.cs
public class Solution {
    public int[][] Merge(int[][] intervals) {
        Array.Sort(intervals, (x,y) => x[0] - y[0]);
        int i = 0, j = 1;
        while (j < intervals.Length) {
            var first = intervals[i];
            var second = intervals[j];
            if (first[1] >= second[0]) {
                first[1] = Math.Max(first[1], second[1]);
            }
            else {
                intervals[i+1] = intervals[j];
                i++;
            }
            j++;
        }
        return intervals[..(i+1)];;
    }
}

File: 0056-merge-intervals/solution.py
class Solution:
    def merge(self, arr: List[List[int]]) -> List[List[int]]:
        arr.sort()
        res = []
        start = arr[0][0]
        end = arr[0][1]
        for i in range(1, len(arr)):
            if end < arr[i][0]:
                res.append([start, end])
                start = arr[i][0]
                end = arr[i][1]
            end = max(end, arr[i][1])
        res.append([start,end])
        return res
